ISSUE #2839 - Revamping - Containers v2 Layout

m1chalkubiak opened the pull request:
This fixes #2839

test url:  https://issue-2839.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

george-hadfield commented:
@m1chalkubiak @sanmont3drepo @carmenfan what is the link for me to view this PR to leave my review?

carmenfan commented:
@george-hadfield as mentioned it's probably good to merge in #2708 and #2796 before we look into this one. but for reference, you should be able to access the page here https://issue-2839.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

george-hadfield commented:
Search isn't working on this PR

![image](https://user-images.githubusercontent.com/78047794/141161238-0e002df6-a5a2-4d9b-bf6f-79c2241b6745.png)

Doesn't let me type at all. Using Chrome Version 95.0.4638.69 (Official Build) (x86_64)

m1chalkubiak commented:
> Search isn't working on this PR
> 
> ![image](https://user-images.githubusercontent.com/78047794/141161238-0e002df6-a5a2-4d9b-bf6f-79c2241b6745.png)
> 
> Doesn't let me type at all. Using Chrome Version 95.0.4638.69 (Official Build) (x86_64)

Thanks @george-hadfield  for pointing this out. Probably it's something that I broke during resolving merge conflicts. As we agree on our yesterday meeting I will work on this task on Monday, adding design changes related to revision details & responsive design changes if those will be ready. I Will let You know when everything on this branch will be ready for review.

m1chalkubiak commented:
/azp run

m1chalkubiak commented:
@george-hadfield 
I've fixed the search mechanism on the containers page and added design changes related to revisions details. 
This part is ready to review.

Still, not sure how about adjusting the project to be more responsive, I saw some new designs on Figma but I have not received any information from You about whether it is accepted yet or not. In addition, referring to these designs, I do not know to what point we should implement them, because they contain a full mobile view, and at this stage of the project, as far as I know, we were not supposed to do that.

sanmont3drepo commented:
I've seen that you changed the header mechanism. Is there a strong reason for this? I feel that before that change the content components for the project and teamspace were more cohesive.

Also the DashboardHeader as defined in this PR is not a general dashboard header but the projectDashboardHeader

m1chalkubiak commented:
> I've seen that you changed the header mechanism. Is there a strong reason for this? I feel that before that change the content components for the project and teamspace were more cohesive.

The main reason are very big changes in the appearance of the dashboard, which were implemented under this PR, and the fact that the navigation does not work properly in the elements rendered by the portal. Personally, I believe that the previous solution was overcomplicated in the context of further development of the application, currently, the whole mechanism has been reduced to single-line 'if' statement

> Also the DashboardHeader as defined in this PR is not a general dashboard header but the projectDashboardHeader

General, I haven't seen the teamspace designs so far, now it made more sense - moved the new header to the project subfolder.

george-hadfield commented:
@m1chalkubiak 

The only thing I can find as an issue (aside from responsive) is the search result text if there are no results. It displays {Search term} instead of the actual search term. 

![image](https://user-images.githubusercontent.com/78047794/142080499-aef2a159-c0fd-4fa9-8057-7bb532b5b109.png)

m1chalkubiak commented:
/azp run

m1chalkubiak commented:
> @m1chalkubiak
> 
> The only thing I can find as an issue (aside from responsive) is the search result text if there are no results. It displays {Search term} instead of the actual search term.
> 
> ![image](https://user-images.githubusercontent.com/78047794/142080499-aef2a159-c0fd-4fa9-8057-7bb532b5b109.png)

@george-hadfield I wasn't able to reproduce it locally, but I guess it could be due to missing translations, I added them and triggered rebuilding the project, hopefully, this will fix the problem.

mosment3DRepo commented:
### Changing page width with scollbard

@m1chalkubiak are you able to fix this issue of the scrollbar changing the page width

See how in the video below when you expand the accordion the content exceeds the page length, which introduces the scroll bar and then shunts everything a few pixels to the left.

https://user-images.githubusercontent.com/62879582/142839403-8880d74a-4c34-4623-b882-291d7b0ba185.mp4

### Current Behaviour
Page changes width and moves content over a few pixels to the left

### Expected Behaviour
Elements should stay in the same place regardless of whether the scrollbar is there or not

m1chalkubiak commented:
> ### Changing page width with scollbard
> @m1chalkubiak are you able to fix this issue of the scrollbar changing the page width
> 
> See how in the video below when you expand the accordion the content exceeds the page length, which introduces the scroll bar and then shunts everything a few pixels to the left.
> 
>  Expand.Collapse.mp4 
> ### Current Behaviour
> Page changes width and moves content over a few pixels to the left
> 
> ### Expected Behaviour
> Elements should stay in the same place regardless of whether the scrollbar is there or not

Of course, I will fix it.

sanmont3drepo commented:
Can we change the tooltip for add favourite when its filled to remove from favourites?
![image](https://user-images.githubusercontent.com/42032433/142884151-8fdfae58-9a34-49dc-9a66-950d8f12a3b1.png)

george-hadfield commented:
@m1chalkubiak is this ready for me to review for responsive? Should I use the test url: https://issue-2839.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers ?

m1chalkubiak commented:
/azp run

m1chalkubiak commented:
Hi @george-hadfield I have just pushed the last changes that I was asked to do and started the branch rebuilding, after it is finished everything should be ready for review.

sanmont3drepo commented on frontend/src/v5/ui/components/dashboard/dashboardList/dashboardListHeaderLabel/dashboardListHeaderLabel.component.tsx:
```diff
@@ -26,19 +26,31 @@ type IDashboardListHeaderLabel = {
 	sort?: boolean;
 	onClick?: Dispatch<void>;
 	width?: number;
+	tabletWidth?: number;
```
This defines the width of the component when it's being displayed in a tablet, right? If that's the case can we rename this property to widthOnTablet? 
Or the other option is making width like this

```suggestion
	width?: number |  DisplaysWidth;
```

where DisplayWitdth is something like  

```
interface DisplayWitdth  {
  default: number;
  tablet?: number;
  desktop?: number;  
}
```
 also that will allow for adding something in the future like 
 
 ```
interface DisplayWitdth  {
  ... 
  bigPhone? : number;
  mediumPhone?: number;
  ... 
}
```

sanmont3drepo commented on frontend/src/v5/ui/components/dashboard/dashboardList/dashboardListItem/components/dashboardListItemText/dashboardListItemText.component.tsx:
```diff
@@ -21,17 +21,29 @@ import { Text } from './dashboardListItemText.styles';
 type IDashboardListItemText = {
 	children: ReactNode;
 	width?: number;
+	tabletWidth?: number;
+	mobileWidth?: number;
 	className?: string;
 	selected?: boolean;
+	hideBelowTablet?: boolean;
```
Take a look at the comments  `dashboardListHeaderLabel.component.tsx `

sanmont3drepo commented on frontend/src/v5/ui/components/dashboard/dashboardList/dashboardListItem/components/dashboardListItemTitle/dashboardListItemTitle.component.tsx:
```diff
@@ -24,16 +24,17 @@ type IDashboardListItemTitle = {
 	children?: ReactNode;
 	subtitle: ReactNode;
 	width?: number;
+	tabletWidth?: number;
```
Take a look at the comments  `dashboardListHeaderLabel.component.tsx `

sanmont3drepo commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListHeaderLabel/revisionsListHeaderLabel.component.tsx:
```diff
@@ -21,8 +21,12 @@ import { Container } from './revisionsListHeaderLabel.styles';
 
 type IDashboardListHeaderLabel = {
 	width?: number;
+	tabletWidth?: number;
+	mobileWidth?: number;
 	className?: string;
 	children?: ReactNode;
+	hideBelowTablet?: boolean;
+	marginRight?: boolean;
 };
 
 export const RevisionsListHeaderLabel = ({ children, ...props }: IDashboardListHeaderLabel) => (
```
I think we could reuse 

`IFixedOrGrowContainer` from  `fixedOrGrowContainer.component.tsx` but export it and rename it to `IFixedOrGrowContainerProps`

Like this

```suggestion
type IDashboardListHeaderLabel = {
	marginRight?: boolean;
} & IFixedOrGrowContainerProps;

export const RevisionsListHeaderLabel = ({ children, ...props }: IDashboardListHeaderLabel) => (
```

sanmont3drepo commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListHeaderLabel/revisionsListHeaderLabel.component.tsx:
```diff
@@ -21,8 +21,12 @@ import { Container } from './revisionsListHeaderLabel.styles';
 	className?: string;
 	children?: ReactNode;
+	hideBelowTablet?: boolean;
+	marginRight?: boolean;
```
Question: in which case would we need marginRight to be true?

m1chalkubiak replied:
I added this extra styling via styles, using the prop for this is actually a bit redundant.

sanmont3drepo commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListItemCode/revisionsListItemCode.styles.ts:
```diff
@@ -14,12 +14,24 @@
+	width: auto;
+	&:after {
+		${({ $active, $hover, theme }) => css`
+			${$active ? fadeToLeft(theme.palette.primary.main) : fadeToLeft(theme.palette.secondary.light)};
```
please let not use words starting with  '$' at the begining as props (or in general).

sanmont3drepo commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListItemText/revisionsListItemText.styles.ts:
```diff
@@ -25,3 +27,7 @@ export const Text = styled(Typography)`
 `;
+
+export const OverflowWrapper = styled(TextOverflow)`
+	width: auto;
```
review if the default witdh of TextOverflow shouldnt be just auto.

sanmont3drepo commented on frontend/src/v5/ui/components/shared/revisionDetails/revisionDetails.styles.ts:
```diff
@@ -29,6 +30,10 @@ export const Container = styled.div`
 	overflow: hidden;
 	overflow-y: scroll;
+
+	@media (max-width: 1024px) {
```
I think we could use a constant for this.

sanmont3drepo commented on frontend/src/v5/ui/controls/fixedOrGrowContainer/fixedOrGrowContainer.component.tsx:
```diff
@@ -21,12 +21,29 @@ import { Container } from './fixedOrGrowContainer.styles';
 
 type IFixedOrGrowContainer = {
 	width?: number;
+	tabletWidth?: number;
+	mobileWidth?: number;
+	hideBelowDesktop?: boolean;
+	hideBelowTablet?: boolean;
```
take a look at the previous comments regarding these types.

sanmont3drepo commented on frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containersList.component.tsx:
```diff
@@ -70,52 +84,99 @@ export const ContainersList = ({
 				tooltipTitles={titleTooltips}
 				isLoading={areStatsPending}
+				sideElement={(
+					<CollapseSideElementGroup>
```
Question: Is it necessary to create this element? Can it be a FixedOrGrowContainer?

sanmont3drepo commented on frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containersList.styles.ts:
```diff
@@ -14,8 +14,31 @@
+			width: 235px;
+		}
+		
+		@media (max-width: 1024px) {
+			max-width: 405px;
+		}
+		
+		@media (min-width: 1025px) {
```
There's a difference of one pixel in these two media queries. 
Was this a copy paste error?
Also we should use constants

sanmont3drepo commented on frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containersList.styles.ts:
```diff
@@ -14,8 +14,31 @@
+	${SearchInputStyles.TextField} {
+		margin-right: 15px;
+		
+		@media (max-width: 768px) {
```
Constant here 😄

m1chalkubiak commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListItemText/revisionsListItemText.component.tsx:
```diff
@@ -17,26 +17,42 @@
+		hideBelowTablet={hideBelowTablet}
+		className={className}
+	>
+		<OverflowWrapper $meta={meta} $hover={hover} $active={active}>
```
Just a way to prevent rendering nonexistent attributes to DOM:
https://styled-components.com/docs/api#transient-props

However, I removed those props that were no longer needed

m1chalkubiak commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListItem/revisionsListItem.component.tsx:
```diff
@@ -42,14 +43,23 @@ export const RevisionsListItem = ({ revision, containerId, active = false }: IRe
 	return (
-		<Container>
-			<RevisionsListItemText meta width={130} active={active}>
+		<Container onMouseOver={toggleHover} onMouseOut={toggleHover}>
```
I changed my approach to adding this color fade effect, now it is done with a mask so there is no need to pass the color which is much more practical.

m1chalkubiak commented on frontend/src/v5/ui/components/shared/revisionDetails/components/revisionsListItem/revisionsListItem.component.tsx:
```diff
@@ -42,14 +43,23 @@ export const RevisionsListItem = ({ revision, containerId, active = false }: IRe
-		<Container>
-			<RevisionsListItemText meta width={130} active={active}>
+		<Container onMouseOver={toggleHover} onMouseOut={toggleHover}>
+			<RevisionsListItemText meta width={130} tabletWidth={94} active={active}>
```
I guess it was just about adding a variant for this component, different typography is used for the dates. Currently, I moved it to a separate component, I think now it's much more clear.

m1chalkubiak commented on frontend/assets/icons/logo.svg.tsx:
```diff
@@ -17,8 +17,8 @@
-export default () => (
-	<svg width="30" height="28" viewBox="0 0 30 28" fill="none" xmlns="http://www.w3.org/2000/svg">
+export default (props) => (
+	<svg width="30" height="28" viewBox="0 0 30 28" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
```
changed

m1chalkubiak commented on frontend/src/v5/ui/components/shared/appBar/appBar.component.tsx:
```diff
@@ -16,33 +16,29 @@
-		<Items>
-			<TopNavigation />
+		</LeftColumn>
+		<RightColumn>
```
At one point in trying to improve the styling of this page element, the styles of these two elements were very different from each other, but in the end I made it more consistent so I went back to the previous split.

m1chalkubiak commented on frontend/src/v5/ui/components/dashboard/dashboardList/dashboardListHeaderLabel/dashboardListHeaderLabel.component.tsx:
```diff
@@ -26,19 +26,31 @@ type IDashboardListHeaderLabel = {
+	tabletWidth?: number;
 	className?: string;
 	name?: string;
+	hideBelowDesktop?: boolean;
```
Done, seems to be a good change.

m1chalkubiak commented on frontend/src/v5/ui/components/shared/breadcrumbs/breadcrumbs.styles.ts:
```diff
@@ -45,9 +47,20 @@ export const Breadcrumb = styled(Button).attrs({
+	}
+`;
+
+export const OverflowWrapper = styled(TextOverflow)`
+	width: auto;
+`;
```
After refactoring, this is the only component where this additional styling is needed.

Charence commented:
@3drepo/product-team Please check this

keifgwinn commented:
/azp run

george-hadfield commented:
I keep getting a "503 Error" when trying check this

https://issue-2839.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

carmenfan commented:
> I keep getting a "503 Error" when trying check this
> 
> https://issue-2839.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

don't worry about this one. The deployments stopped working due to changes made by @3drepo/devops .

@sanmont3drepo  need to resolve conflicts before we can deploy again. And he is going to wait on https://github.com/3drepo/3drepo.io/pull/2850 to be merged in first.

I just triggered a build for https://github.com/3drepo/3drepo.io/pull/2850 , it should be up in ~10mins

sanmont3drepo commented:
This needs to be addressed here: 
https://github.com/3drepo/3drepo.io/pull/2850#issuecomment-989902336

sanmont3drepo commented:
/azp run

george-hadfield commented:
### Gradient appears too early in containers list
There is still sufficient space to show the title, there is no need for a gradient here

![image](https://user-images.githubusercontent.com/78047794/146193296-8e13b20c-5375-4660-b714-220afabf081a.png)

george-hadfield commented:
@sanmont3drepo I can still see the gradient here. I just noticed that this PR was waiting my review but the last comment I made 20 days isn't fixed. Am I looking at the right link, or is this expected?

![image](https://user-images.githubusercontent.com/78047794/148067439-042bded4-bcb9-4534-af93-4e7282c73e18.png)

carmenfan commented:
> v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

this was spun out to a different issue. you can see it's already addressed in staging https://staging.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

carmenfan commented:
/azp run

carmenfan commented:
> > v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers
> 
> this was spun out to a different issue. you can see it's already addressed in staging https://staging.dev.3drepo.io/v5/dashboard/carmen/843ba9f0-00c6-11ec-8d97-41a278fb55fd/t/containers

@george-hadfield  nvm i thought this was the shadow issue

@sanmont3drepo  i think you missesd something here

sanmont3drepo commented:
/azp run

carmenfan commented:
@george-hadfield  ready for another look
